core/src/main/java/com/yahoo/imapnio/async/internal/ImapAsyncSessionImpl.java

```diff
@@ -261,6 +266,13 @@ public void recordResponseBytes(final int length) {
             responseTotalBytes += length;
         }
 
+        /**
+         * @return request start time.
+         */
+        public long getRequestStartTimeInMillis() {
```

Comment:
what is the reason of not using requestSentTime which is already there?

Reply:
Fan, is this the time recorded when the command submitting to the server?

Reply:
Looks like requestSentTime doesn't really point to the time time when request is being sent. Had checked with Vikram as well.

Reply:
I see it is fine then. keep the requestStartTimeInMillis, I will delete requestSentTime.

Reply:
Change line 241 to append(getRequestStartTimeInMillis()) as well.